tests/test_cursor_common_async.py

```diff
@@ -900,3 +900,20 @@ async def test_error_no_result(aconn):
         psycopg.ProgrammingError, match="last operation.*result status: EMPTY_QUERY"
     ):
         await cur.fetchone()
+
+
+async def test_row_maker_returns_none(aconn):
+    cur = aconn.cursor(row_factory=rows.scalar_row)
+    query = "values (null), (0)"
+    recs = [None, 0]
+
+    await cur.execute(query)
+    assert [await cur.fetchone() for _ in range(len(recs))] == recs
+    await cur.execute(query)
+    assert await cur.fetchmany(len(recs)) == recs
+    await cur.execute(query)
+    assert await cur.fetchall() == recs
+    await cur.execute(query)
+    assert [row async for row in cur] == recs
+    stream = cur.stream(query)
+    assert [row async for row in stream] == recs
```

Comment:
We have an `alist` helper you can use here:
```suggestion
    assert await alist(cur) == recs
    assert await alist(cur.stream(query)) == recs
```
which will be converted to `list` by async_to_sync.

Reply:
Done.